### PR TITLE
[ENH] Adding intervention graph classes for known graphs and equivalence classes and improve graph class documentation

### PR DESCRIPTION
**Pull request by adam2392**
Signed-off-by: Adam Li <adam2392@gmail.com>

Xref PR implementing the PsiFCI algorithm: https://github.com/py-why/dodiscover/pull/111

Changes proposed in this pull request:

- Implements `AugmentedGraph`, `IPAG` and `PsiPAG` with minimal code-duplication
- Adds unit tests on the F-nodes
- Adds documentation details to ADMG, CPDAG and PAG that addresses some of the shortcomings described in https://github.com/py-why/dodiscover/issues/114
- the main file to review is `pywhy_graphs/classes/intervention.py`

## Before submitting

- [ ] I've read and followed all steps in the [Making a pull request](https://github.com/py-why/pywhy-graphs/blob/main/CONTRIBUTING.md#making-a-pull-request)
    section of the `CONTRIBUTING` docs.
- [ ] I've updated or added any relevant docstrings following the syntax described in the
    [Writing docstrings](https://github.com/py-why/pywhy-graphs/blob/main/CONTRIBUTING.md#writing-docstrings) section of the `CONTRIBUTING` docs.
- [ ] If this PR fixes a bug, I've added a test that will fail without my fix.
- [ ] If this PR adds a new feature, I've added tests that sufficiently cover my new functionality.

## After submitting

- [ ] All GitHub Actions jobs for my pull request have passed.

### CODE REVIEW

File: pywhy_graphs/classes/intervention.py [hunk 1]
```diff
@@ -1,11 +1,350 @@
...
+                "There is a graph property named F-nodes already that is not of type dict."
+            )
+
+    def add_f_node(self, intervention_set: Set[Node]):
+        """Add an F-node to the graph.
+
+        Parameters
+        ----------
+        intervention_set : Set[Node]
+            A set of regulard nodes that already exist in the causal graph.
```
Comment (jaron-lee) [inline pywhy_graphs/classes/intervention.py:29]: Small typo here?

File: pywhy_graphs/classes/intervention.py [hunk 2]
```diff
@@ -1,11 +1,350 @@
...
+            nodes = nodes.union(iset)
+        return nodes
+
+
+class AugmentedGraph(ADMG, InterventionMixin):
+    """An augmented ADMG causal graph.
+
+    An augmented graph is one where interventions are represented by F-nodes.
+    See :footcite:`pearl_aspects_1993`, where they were first introduced. They
+    allow one to
```
Comment (jaron-lee) [inline pywhy_graphs/classes/intervention.py:97]: and possibly here?

File: pywhy_graphs/classes/intervention.py (hunk 1 above)
Comment (adam2392) [reply pywhy_graphs/classes/intervention.py:29]: Fixed

**Comment by adam2392** [comment]
Docs: https://output.circle-artifacts.com/output/job/9301c837-88b2-4f0a-9390-1d778fbc2be1/artifacts/0/dev/use.html

**Comment by adam2392** [comment]
User-guide: https://output.circle-artifacts.com/output/job/cd549827-c263-4343-a1d7-12dc9455ae5e/artifacts/0/dev/user_guide.html

Intervention graphs: https://output.circle-artifacts.com/output/job/cd549827-c263-4343-a1d7-12dc9455ae5e/artifacts/0/dev/generated/pywhy_graphs.AugmentedGraph.html#pywhy_graphs.AugmentedGraph

**Comment by adam2392** [comment]
Whenever you guys get the chance, if either of you are familiar enough w/ the intervention graphs to provide a review, it would be very helpful!

The corresponding implementation of PsiFCI with known and unknown intervention targets is implemented in https://github.com/py-why/dodiscover/pull/111

**Comment by adam2392** [comment]
@jaron-lee and @robertness lmk if you guys prefer to take a review before I merge. Just cuz I need it in `main` for dodiscover Psi-FCI PR. 

However, think some reviews is good at least to get the team acquainted

**Comment by adam2392** [comment]
Ima go ahead and merge this to get PsiFCI working.

**Pull request merged**